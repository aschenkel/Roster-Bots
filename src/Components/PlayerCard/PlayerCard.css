.card {
  background-color:wheat;
  height:180px;
  margin: 16px 0px 16px 0px;
  min-width: 400px;
  max-width: 400px;
  box-shadow: 0 8px 12px 0 rgba(0,0,0,0.2);
  display: inline-block;
  backface-visibility: hidden;
  -webkit-backface-visibility: hidden;
  overflow: hidden;
}

.card:hover{
  transform: scale3d(1.1,1.1,1.1);
  -ms-transform: scale3d(1.1,1.1,1.1);
  -webkit-transform: scale3d(1.1,1.1,1.1);
}
.badge{
    background-color: salmon;
    width: 30px;
    height: 35px;
    border-radius: 50%;
    padding: 8px;
    color: white

}
.atributes{
  display: flex;
  margin: 20px 0 0 12px 
}
.content{
  margin: 16px
}
.name{
    font-size: 18px;
    font-weight: bold;
    display: block;
    width: 350px;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    color: #1C0113;
    margin: -8px 0 0 0 
}

.name-tag{
    font-size: 14px;
    display: block;
    text-shadow: 0px 1px 0px rgba(255,255,255,.3), 0px -1px 0px rgba(0,0,0,.7);
    color:gray
}
.atributes-text{
    color:gray;
    text-shadow: 0px 1px 0px rgba(255,255,255,.3), 0px -1px 0px rgba(0,0,0,.7);

}
.speed-separators{
  margin: 0px 0 20px 0
}
.speedBadge{
  margin: 0 0 0 -20px
}
.icons{
  margin: 0px 8px 0 0;
  color:gray
}

p{
  font-size: 20px;
  white-space: nowrap;
  display: inline;

}

.values{
  margin: 0 0 0 16px
}

p.score{
  font-size: 100px;
  float: right;
  font-weight: bold;
  margin: -10px 0 0 95px;
	color: #EED8AE;
  text-shadow: 0px 4px 0px  rgba(255,255,255,.3), 0px -2px 0px grey;
}

p.score-exception{
  font-size: 100px;
  float: right;
  font-weight: bold;
  margin: -10px 0 0 45px;
	color: #EED8AE;
  text-shadow: 0px 4px 0px  rgba(255,255,255,.3), 0px -2px 0px grey;
}


@media only screen and (max-width: 900px) {
    .card {
        width: 60%;
        margin: 16px 25%;
    }
}