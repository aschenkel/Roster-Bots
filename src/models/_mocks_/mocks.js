import {numberOfPlayersPerTeam,maxScoreAllowed, numberOfStarters} from '../../config/rules'


export const createMockPlayer = (score) =>{
    var halfScore = score/2
    return {
        name:mockNameCreator(),
        speed: Math.round(halfScore),
        strenght:Math.floor(halfScore),
        score
    }
}

export const mockNameCreator = () => {
    return '081813ad-c751-40e6-a12b-04f61968607b'
}

export const createMockRoster = () =>{
    var mockPlayers = []
    for (var i = 0; i < numberOfPlayersPerTeam; i++) {
      var properScore = maxScoreAllowed - i;
      mockPlayers.push(createMockPlayer(properScore))
    }
    return mockPlayers
}


