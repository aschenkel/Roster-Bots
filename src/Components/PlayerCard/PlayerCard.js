import React from 'react';
import PropTypes from 'prop-types';
import './PlayerCard.css';
import PlayerName from './PlayerName/PlayerName'
import PlayerAttribute, {attributes} from './PlayerAttribute/PlayerAttribute'
import Score from './Score/Score'
//       

const PlayerCard = (props) => {
    return (
      <div className="card">
        <div className="data">
          <div className="name-container">
            <PlayerName name={props.player.name}/>
          </div>
          <div className="atributes">
                <PlayerAttribute attributeType= {attributes.speed} value={props.player.speed}/>
                <PlayerAttribute attributeType = {attributes.strenght} value={props.player.strenght}/>
          </div>
        </div>
        <div className="player-score">
         <Score score={props.player.score}/>
        </div>
      </div>
    );
}


PlayerCard.propTypes = {
  player: PropTypes.object.isRequired
};

PlayerCard.defaultProps = {
  displayName: 'PlayerCard'
}

export default PlayerCard;