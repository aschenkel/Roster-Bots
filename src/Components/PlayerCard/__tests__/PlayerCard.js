import React from 'react';
import PlayerCard from '../PlayerCard';
import renderer from 'react-test-renderer';
import {createMockPlayer} from '../../../models/_mocks_/mocks'

it('should render a player card', () => {
    var mockPlayer = createMockPlayer()
    const tree = renderer.create(
        <PlayerCard player={mockPlayer}/>
        ).toJSON();
    expect(tree).toMatchSnapshot();
});

