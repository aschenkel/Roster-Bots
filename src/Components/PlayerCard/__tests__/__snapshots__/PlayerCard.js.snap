// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a player card 1`] = `
<div
  className="card"
>
  <div
    className="data"
  >
    <div
      className="name-container"
    >
      <div
        className="content"
      >
        <p
          className="name-tag"
        >
          name
        </p>
        <p
          className="name"
        >
          081813ad-c751-40e6-a12b-04f61968607b
        </p>
      </div>
    </div>
    <div
      className="atributes"
    >
      <div>
        <p
          className="badge"
        >
          NaN
        </p>
        <p
          className="atributes-text"
        >
          speed
        </p>
      </div>
      <div>
        <p
          className="badge"
        >
          NaN
        </p>
        <p
          className="atributes-text"
        >
          strenght
        </p>
      </div>
    </div>
  </div>
  <div
    className="player-score"
  >
    <p
      className="score"
    >
       
    </p>
  </div>
</div>
`;
